e2e/tests/test_cases/node_disconnection_test.robot

```diff
@@ -0,0 +1,126 @@
+*** Settings ***
+Documentation    Node disconnection test
+...              https://github.com/longhorn/longhorn/issues/1545
+
+Test Tags    manual_test_case
+
+Resource    ../keywords/common.resource
+Resource    ../keywords/volume.resource
+Resource    ../keywords/setting.resource
+Resource    ../keywords/network.resource
+Resource    ../keywords/statefulset.resource
+Resource    ../keywords/workload.resource
+
+Test Setup    Set test environment
+Test Teardown    Cleanup test resources
+
+*** Variables ***
+${LOOP_COUNT}    3
+${RETRY_COUNT}    300
+${RETRY_INTERVAL}    1
+
+*** Test Cases ***
+Node Disconnect And Keep Data Writing And No Replica On The Disconnected Node
+    [Documentation]    -- Manual test plan --
+    ...                Disable auto-salvage.
+    ...                Create a volume with 2 replicas on the 1st and the 2nd node.
+    ...                Attach the volume to the 3rd node.
+    ...                Keep writing data directly to the volume. (sudo dd if=/dev/urandom of=/dev/longhorn/test-1 bs=2M count=2048)
+    ...                Disconnect the network of the node that the volume attached to for 100 seconds during the data writing. (sudo nohup ./network_down.sh 100)
+    ...                Wait for the node back.
+    ...                The volume should remain detached, and all replicas remain in ERROR state.
+    Given Set setting auto-salvage to false
+
+    When Create volume 0 with 10 GB and no replicas on the attached node
+    And Keep writing data to volume 0
+    And Disconnect volume 0 node network for 100 seconds without waiting for completion
+
+    Then Wait for volume 0 detached
+    And Wait for volume 0 faulted
+
+Node Disconnect And Keep Data Writing And Have Replica On The Disconnected Node
+    [Documentation]    -- Manual test plan --
+    ...                Disable auto-salvage.
+    ...                Create a volume with 3 replicas.
+    ...                Attach the volume to a node.
+    ...                Keep writing data directly to the volume. (sudo dd if=/dev/urandom of=/dev/longhorn/test-1 bs=2M count=2048)
+    ...                Disconnect the network of the node that the volume attached to for 100 seconds during the data writing. (sudo nohup ./network_down.sh 100)
+    ...                Wait for the node back.
+    ...                The volume will be in degraded state and then started the replica rebuilding process.
+    ...                The volume become healthy.
+    Given Set setting auto-salvage to false
+
+    When Create volume 0 with 10 GB and 3 replicas
+    And Attach volume 0 to node 1
+    And Wait for volume 0 healthy
+
+    When Keep writing data to volume 0
+    And Disconnect volume 0 node network for 100 seconds without waiting for completion
+
+    Then Wait for volume 0 degraded
+    And Wait for volume 0 healthy
```

Comment:
Can we check:
```
Wait for the node back.
```
like the original manual test case?

Reply:
Updated